app/src/main/java/com/p2p/presentation/tuttifrutti/TuttiFruttiViewModel.kt

```diff
@@ -18,4 +22,70 @@ class TuttiFruttiViewModel(
     bluetoothConnectionCreator,
     instructionsRepository,
     Game.TUTTI_FRUTTI
-)
+) {
+
+    /**Data for all game*/
+    private val roundsInfo = mutableListOf<RoundInfo>()
+
+    private val lettersByRound: List<Char> by lazy { getRandomLetters() }
+
+    private val _totalRounds = MutableLiveData<Int>()
+    val totalRounds: LiveData<Int> = _totalRounds
+
+    private val _selectedCategories = MutableLiveData<List<Category>>()
+    val selectedCategories: LiveData<List<Category>> = _selectedCategories
+
+
+    /**Data for actual round*/
+    private val _actualRound = MutableLiveData<Int>()
+    val actualRound: LiveData<Int> = _actualRound
+    private val _actualLetter = MutableLiveData<Char>()
+    val actualLetter: LiveData<Char> = _actualLetter
+
+
+    /**Before playing game*/
+    fun setSelectedCategories(categories: List<Category>) { _selectedCategories.value = categories }
+
+    fun setTotalRounds(totalRounds: Int) { _totalRounds.value = totalRounds }
+
+    fun startGame() {
+        _actualRound.value = 1
+        _actualLetter.value = lettersByRound[0]
+    }
+
+    private fun getRandomLetters(): List<Char> =
+        availableLetters.map { it }.shuffled().subList(0, totalRounds.value!!)
+
+
+    /**On Playing game*/
+    fun finishRound(categoriesWithValues: Map<Category, String>) {
+        if (gameContinues()) {
+            generateNextRoundValues()
+        }
+        roundsInfo.add(RoundInfo(actualLetter.value!!, categoriesWithValues))
+        goToReview()
```

Comment:
```suggestion
        goToReviewOrWait()
```
A la review solo va el server, los clientes tienen que esperar.

Reply:
Nais!